src/component/mxgraph/MxGraphRenderer.ts

```diff
@@ -156,10 +163,22 @@ export default class MxGraphRenderer {
     }
     return mxCell;
   }
+
+  public refreshCell(bpmnElementId: string): void {
+    const mxCell = this.graph.getModel().getCell(bpmnElementId);
+    if (!mxCell) {
```

Comment:
⚠️ IMHO this is not the right place for this function
A lot of code has been modified to make the move work.
First, about having to inject the CssRegistry: we may have kept it as argument to avoid all the changes.
Second, except that we manipulate mxgraph objects here, the initial mxgraph model creating and the individual cells update have no relation. In addition, we now have a circle dependency between the mxgraph classes and the registry classes.
We could create a dedicated class in charge of triggering/managing the mxGraph model/state which could send event if we want, or direcly change the cell state as we do for now.

Reply:
For me, it's the better place, because we touch to the mxGraph rendering. But I agree, there is a lot of configuration. This is why I wanted to introduce **'JS Event'** to keep the different parts of the library separate.

Reply:
I agree with @csouchet  that it is better place as by aspect it is rendering oriented action.
As @tbouffard noticed it is definitely coupled to tight now and we can do better

Using Event can be a good approach :)
maybe cssRegistry upon adding classes can emit an Event upon which Renderer can do the 'refreshCell'

Reply:
I didn't see how to do that before, but I remove the CssRegistry from the MxGraphRenderer.
For the events, we though with @tbouffard to do that in another refatoring PR.
Maybe, we want to use events for other features.